src/js/d3ViewComponent.js

```diff
@@ -114,11 +139,92 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
     };
 
     // Returns a formatted string for a numeric data value based on a supplied template
-    floe.d3ViewComponent.getTemplatedDisplayValue = function(totalValue, percentageDigits, template, d) {
+    floe.d3ViewComponent.getTemplatedDisplayValue = function (totalValue, percentageDigits, template, d) {
         var percentage = floe.chartAuthoring.percentage.calculate(d.value, totalValue);
         var percentageForTemplate = percentage !== null ? percentage.toFixed(percentageDigits) : percentage;
         var output = fluid.stringTemplate(template, {label: d.label, value: d.value, percentage: percentageForTemplate, total: totalValue});
         return output;
     };
 
+    // Given a data key used to maintain object constancy in D3, a DOM
+    // element with a unique ID and the component, updates the key's
+    // value (an array of IDs) to include that ID
+    floe.d3ViewComponent.addElementIdToDataKey = function (d3Key, idToAdd, that) {
+        var keyPath = ["dataKeys", d3Key, idToAdd];
+        that.applier.change(keyPath, true);
+    };
+
+    // Corresponding "remove" functionality to addElementIdToDataKey
+    floe.d3ViewComponent.removeElementIdFromDataKey = function (d3Key, idToRemove, that) {
+        var keyPath = ["dataKeys", d3Key, idToRemove];
+        that.applier.change(keyPath, false, "DELETE");
+    };
+
+    // Given an object "elements" with element IDs as keys, returns a
+    // joined string of IDs suitable for use as a jQuery selector to select all
+    // those IDs
+    floe.d3ViewComponent.getElementIdsAsSelector = function (elementIds) {
+        if (fluid.isPlainObject(elementIds)) {
+            var keys = Object.keys(elementIds);
+            var elemIdCollectionWithPreface = fluid.transform(keys, function (elemId) {
+                    return "#" + elemId;
```

Comment:
This is not reliable in general (although it is for our ids) because of escaping rules for characters in selectors.
Rather than fabricate a selector, it would be much more direct to simply apply

var elements = $(fluid.transform(keys, fluid.byId)); 

at the point where the elements are wanted